frontends/web/src/routes/account/manage/manage.tsx

```diff
@@ -0,0 +1,221 @@
+import { Entry } from '../../../components/guide/entry';
+import { Guide } from '../../../components/guide/guide';
+import { Header } from '../../../components/layout';
+import { translate, TranslateProps } from '../../../decorators/translate';
+import { Step, Steps } from './components/steps';
+import { CoinDropDown } from './components/coin-dropdown';
+import * as styles from '../manage/manage.css';
+import checkicon from '../../../assets/icons/check.svg';
+
+interface AddAccountProps {
+    // type?: 'multi' | 'btconly'
+}
+
+type Props = AddAccountProps & TranslateProps;
+
+interface State {
+    accountName: string;
+    coinCode: 'choose' | accountApi.CoinCode;
+    onlyOneSupportedCoin: boolean;
+    step: 0 | 1 | 2;
+    supportedCoins: string[];
+}
+
+class ManageAccount extends Component<Props, State> {
+    public readonly state: State = {
+        accountName: '',
+        coinCode: 'choose',
+        onlyOneSupportedCoin: false,
+        step: 0,
+        supportedCoins: [],
+    };
+
+    public componentDidMount() {
+        getSupportedCoins()
+            // TEST with only 1 coin
+            // .then(() => (['btc']))
+            .then((coins) => {
+                const onlyOneSupportedCoin = (coins.length === 1);
+                this.setState({
+                    // @ts-ignore
+                    coinCode: onlyOneSupportedCoin ? coins[0] : 'choose',
+                    onlyOneSupportedCoin,
+                    supportedCoins: coins
+                });
+            })
+            .catch(console.error);
+    }
+
+    private getStep = () => {
+        const { onlyOneSupportedCoin, step } = this.state;
+        if (onlyOneSupportedCoin) {
+            return step === 0 ? 'choose-name' : 'success';
+        }
+        switch (step) {
+            case 0: return 'select-coin';
+            case 1: return 'choose-name';
+            case 2: return 'success';
+        }
+    }
+
+    private back = () => {
+        this.setState(({ step }) => {
+            switch (step) {
+                case 0:
+                case 1: return ({ step: 0 });
+                case 2: return ({ step: 1 });
+            }
+        });
+    }
+
+    private next = () => {
+        const { accountName, coinCode } = this.state;
+        switch (this.getStep()) {
+            case 'select-coin':
+                console.info(`${coinCode} selected`);
+                break;
+            case 'choose-name':
+                // TODO: post accountName and coinCode to backend
+                console.info(`add new account for ${coinCode} with the name: ${accountName}`);
+                break;
+            case 'success':
+                route('/account-summary');
+                return;
+                break;
+        }
+        this.setState((state) => {
+            switch (state.step) {
+                case 0: return ({ step: 1 });
+                case 1: return ({ step: 2 });
+                case 2: return ({ step: 2 });
```

Comment:
go from 2 to 2? :thinking:

Reply:
Yeah I know it is silly and not necessary, but else Typescript complains about `Not all code paths return a value.ts(7030)`

Reply:
In https://github.com/digitalbitbox/bitbox-wallet-app/pull/1258 I saw that this state transition has to happen after a successful apiPost anyway, but: you could add a `default:` case that does nothing instead.

Reply:
fixed in https://github.com/digitalbitbox/bitbox-wallet-app/pull/1277